ta/pkcs11/src/processing_symm.c

```diff
@@ -751,6 +857,73 @@ enum pkcs11_rc step_symm_operation(struct pkcs11_session *session,
 
 		break;
 
+	case PKCS11_CKM_MD5_HMAC_GENERAL:
+	case PKCS11_CKM_SHA_1_HMAC_GENERAL:
+	case PKCS11_CKM_SHA224_HMAC_GENERAL:
+	case PKCS11_CKM_SHA256_HMAC_GENERAL:
+	case PKCS11_CKM_SHA384_HMAC_GENERAL:
+	case PKCS11_CKM_SHA512_HMAC_GENERAL:
+		assert(session->processing->extra_ctx);
+		hmac_len = *(uint32_t *)session->processing->extra_ctx;
+
+		switch (function) {
+		case PKCS11_FUNCTION_SIGN:
+			if (out_size < hmac_len) {
+				/* inform client of required size */
+				out_size = hmac_len;
+				output_data = true;
+				rc = PKCS11_CKR_BUFFER_TOO_SMALL;
+				goto out;
+			}
+
+			res = TEE_MACComputeFinal(proc->tee_op_handle,
+						  in_buf, in_size,
+						  computed_mac,
+						  &computed_mac_size);
+			if (res == TEE_SUCCESS) {
+				/* truncate to hmac_len */
+				memcpy(out_buf, computed_mac, hmac_len);
```

Comment:
TEE_MemMove() is the one we probably should be using here.

Reply:
ok